src/posts/controllers/posts.controller.ts

```diff
@@ -8,38 +8,78 @@ export class PostsController {
   constructor(private readonly postsService: PostsService) {}
 
   @Get()
-  findAll() {
-    return this.postsService.findAll();
+  async findAll() {
+    try {
+      return await this.postsService.findAll();
+    } catch (error) {
+        throw new NotFoundException(error); 
+    }
```

Comment:
- Como na linha 13 vc já tá  dando um `return`, o `await` passa a ser desnecessário, então se preferir pode remover
- Caso `postService.findAll` não encontre nenhum registro no banco de dados, ele irá retornar um array vazio, então nesse caso, ele não cai no `catch`. Aí na linha 15 acho que a gente poderia usar outra exception, tipo um internal error:
(o mesmo se aplica pro `findAllAdmin`, `filter` e `findOne`)

```suggestion
  async findAll() {
    try {
      return this.postsService.findAll();
    } catch (error) {
        throw new InternalServerErrorException(error); 
    }
```